types/openlayers/index.d.ts

```diff
@@ -14042,7 +14043,7 @@ declare module olx {
             hidpi?: boolean;
             logo?: (string | olx.LogoOptions);
             tileGrid?: ol.tilegrid.TileGrid;
-            projection: ol.ProjectionLike;
+            projection?: ol.ProjectionLike;
```

Comment:
`ProjectionLike` already have `undefined` in its type definition. Is there any reason for this optional?
And if you want to change this, IMO, you need to change other codes using `ol.ProjectionLike` for consistency...

Reply:
If `ProjectionLike` is not optional, the `undefined` value has to be explicit. A user would have to say `projection: undefined`, which is not ideal.

Would it make more sense for someone to post a separate PR to make all of the `ProjectionLike` instances optional? Changing them all in this PR might clutter the purpose of this one.

Reply:
@iamthechad IMO, including this in that 'separate PR' is better in terms of consistency, and to make less confusion.

Reply:
@Ailrun Agreed.

Reply:
@Ailrun I initially considered changing all the references to optional, but after I checked the source code it looks like in some cases it is not optional at all.

I think the documentation is wrong in this case, because the code clearly expects a projection, but the docs mention ProjectionLike which includes undefined.

This is why I decided to apply the optional to just this case (where it is indeed optional). I'm not familiar enough with the library to know which other cases fit this.

What do you think we should do? 
I could change all references, but then people might make mistakes ...

Reply:
`TileWMS` just pass its projection (whose type is `ProjectionLike`) to `TileImage`, and to `Tile`, ...
https://github.com/openlayers/openlayers/blob/f6eeee37d2e4b13f9ff65acb6b671d9d4c35b505/src/ol/source/tilewms.js#L43
so there's no reason for only `TileWMS` having different type.